internal/handlers/pg/msg_explain.go

```diff
@@ -84,7 +84,9 @@ func (h *Handler) MsgExplain(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 	var port int32
 	connInfo := conninfo.GetConnInfo(ctx)
 	if connInfo.PeerAddr != nil {
-		port = int32(connInfo.PeerAddr.(*net.TCPAddr).Port)
+		if tcpAddr, ok := connInfo.PeerAddr.(*net.TCPAddr); ok {
```

Comment:
In the case when `ok == false` the port is `0`, and then we consider the connection uses a socket, right?

Reply:
Yeah. Also I didn't change returned document structure. The returned document will have "port":0 for unix socket connection.